utils/src/utils/AzExtFsExtra.ts

```diff
@@ -78,7 +78,15 @@ export namespace AzExtFsExtra {
 
     export async function readJSON<T>(resource: Uri | string): Promise<T> {
         const file = await readFile(resource);
-        return JSON.parse(file) as T;
+        try {
+            JSON.parse(file)
```

Comment:
This is parsing twice but doesn't need to. You can do this instead:

```typescript
try {
    return JSON.parse(file) as T;
} catch (err) {
    const pError = parseError(err);
    if (pError.errorType === 'SyntaxError') {
        // If it's a syntax error, wrap it in a better error message
        throw new Error(`Error parsing JSON file: ${resource}. ${pError.message}`);
    } else {
        // Otherwise rethrow the original error
        throw err;
    }
}